File: music-playback/src/main/java/mobile/substance/sdk/music/playback/HeadsetPlugReceiver.java
package mobile.substance.sdk.music.playback;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Created by Julian Os on 30.01.2016.
 */
public class HeadsetPlugReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        PlaybackRemote.INSTANCE.pause();
    }
}

File: music-playback/src/main/java/mobile/substance/sdk/music/playback/HeadsetPlugReceiver.kt
/*
 * Copyright 2016 Substance Mobile
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package mobile.substance.sdk.music.playback

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter

/**
 * Audio noisy receiver. register() and unregister() to enable and disable this. For example, if you are currently playing music this should be registered
 */
object HeadsetPlugReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) = PlaybackRemote.pause()

    infix fun register(context: Context) = context.registerReceiver(this, IntentFilter(Intent.ACTION_HEADSET_PLUG))

    infix fun unregister(context: Context) = context.unregisterReceiver(this)
}